devices/siglis.js

```diff
@@ -4,34 +4,108 @@ const exposes = require('zigbee-herdsman-converters/lib/exposes');
 const reporting = require('zigbee-herdsman-converters/lib/reporting');
 const e = exposes.presets;
 
+const siglisManufacturerCode = 0x129C;
+const zigfredEndpoint = 5;
+
+const zifgredFromZigbee = {
+    cluster: 'manuSpecificSiglisZigfred',
+    type: ['attributeReport'],
+    convert: (model, msg, publish, options, meta) => {
+        const buttonEvent = msg.data['buttonEvent'];
+
+        if (buttonEvent != null) {
+            const buttonLookup = {
+                0: 'button_1',
+                1: 'button_2',
+                2: 'button_3',
+                3: 'button_4',
+            };
+
+            const actionLookup = {
+                0: 'release',
+                1: 'click',
+                2: 'double',
+                3: 'hold',
+            };
+
+            const button = buttonEvent & 0xff;
+            const state = (buttonEvent >> 8) & 0xff;
+
+            const buttonName = buttonLookup[button];
+            const stateName = actionLookup[state];
+
+            if (buttonName && stateName) {
+                const action = `${buttonName}_${stateName}`;
+                return {action};
+            }
+        }
+    },
+};
 
 module.exports = [
     {
         zigbeeModel: ['zigfred uno'],
         model: 'ZFU-1D-CH',
         vendor: 'Siglis',
         description: 'zigfred uno smart in-wall switch',
-        exposes: [e.light_brightness_colorxy().withEndpoint('l1'), e.switch().withEndpoint('l2'), e.light_brightness().withEndpoint('l3')],
-        fromZigbee: [fz.color_colortemp, fz.on_off, fz.brightness, fz.level_config, fz.power_on_behavior, fz.ignore_basic_report],
-        toZigbee: [tz.light_onoff_brightness, tz.light_color, tz.ignore_transition, tz.ignore_rate, tz.light_brightness_move,
-            tz.light_brightness_step, tz.level_config, tz.power_on_behavior, tz.light_hue_saturation_move,
-            tz.light_hue_saturation_step, tz.light_color_options, tz.light_color_mode],
+        exposes: [e.light_brightness_colorxy().withEndpoint('l1'), e.switch().withEndpoint('l2'), e.light_brightness().withEndpoint('l3'),
+            e.action([
+                'button_1_click', 'button_1_hold', 'button_1_release', 'button_2_click', 'button_2_hold', 'button_2_release',
+                'button_3_click', 'button_3_hold', 'button_3_release', 'button_4_click', 'button_4_hold', 'button_4_release'])],
```

Comment:
Please rename `_click` to `_single` (to bring it more inline how other devices work)